tests/articles/viewArticle/viewArticleCreatedByAnotherUser.spec.js

```
@@ -1,25 +1,62 @@
-import { createArticle } from '../../../src/ui/actions/articles/createArticle';
-import { signUpUser } from '../../../src/ui/actions/auth/signUpUser';
+import { generateStorageStateForAuth } from '../../../src/common/helpers/generateStorageStateForAuth';
 
-test.use({ contextsNumber: 2, usersNumber: 2 });
+test.use({ usersNumber: 2 });
 
-test.beforeEach(async ({ pages, users, articleWithoutTags }) => {
-  await signUpUser(pages[0], users[0], 1);
-  await signUpUser(pages[1], users[1], 2);
-  await createArticle(pages[0], articleWithoutTags, 1);
+test.beforeEach(async ({
+  articlesApi,
+  registeredUsers,
+  articleWithoutTags,
+}) => {
+  const author = registeredUsers[0];
+
+  const articlePayload = {
+    title: articleWithoutTags.title,
+    description: articleWithoutTags.description ?? 'Article created via API',
+    body: articleWithoutTags.text,
+    tagList: [],
+  };
+
+  const response = await articlesApi.createArticle(
+    articlePayload,
+    author.token,
+  );
+
+  expect(response.status()).toBe(200);
+
+  const slug = await articlesApi.parseSlugFromResponse(response);
+  articleWithoutTags.url = `/article/${slug}`;
 });
 
 test('View an article created by another registered user', async ({
+  browser,
+  registeredUsers,
   articleWithoutTags,
-  pages,
-  users,
 }) => {
-  const page = new InternalViewArticlePage(pages[1], 2);
+  const author = registeredUsers[0];
+  const viewer = registeredUsers[1];
 
+  // generateStorageStateForAuth MAY be async → we await it safely.
+  const viewerStorage = await generateStorageStateForAuth(viewer);
+
+  // Correct logic: If helper already returns { storageState: ... } keep it.
+  // Otherwise wrap raw cookie JSON as { storageState: <json> }
+  const contextOptions =
+    viewerStorage && typeof viewerStorage === 'object' && 'storageState' in viewerStorage
+      ? viewerStorage
+      : { storageState: viewerStorage };
```

If `generateStorageStateForAuth` returns a falsy value (`undefined`/`null`) this will produce `{ storageState: undefined }` and may cause `browser.newContext` to throw. Add a safer fallback: use the `{ storageState: viewerStorage }` branch only when `viewerStorage` is truthy, otherwise pass `{}` or throw a clear error so tests fail fast with a helpful message.